### PR TITLE
plugins/undotree: switch to settings-style

### PR DESCRIPTION
**Pull request by GaetanLepage**

### CODE REVIEW

File: plugins/utils/undotree.nix [hunk 1]
```diff
@@ -13,120 +13,55 @@ with helpers.vim-plugin;
...
+    # TODO introduced 2024-02-19: remove 2024-03-19
+    imports = let
+      basePluginPath = ["plugins" "undotree"];
+    in
+      map
+      (
+        optionName:
+          mkRenamedOptionModule
+          (basePluginPath ++ [optionName])
+          (basePluginPath ++ ["settings" optionName])
```
Comment (traxys) [inline plugins/utils/undotree.nix:25]: The renamed options won't have the right name I think, they need  to start by an uppercase letter

**Review by traxys — COMMENTED** [review]
I think before removing the options we need a way to link to the plugin homepage, or at the minimum add a `meta.url` option in order to fetch it from the documentation.

File: plugins/utils/undotree.nix [hunk 2]
```diff
@@ -13,120 +13,55 @@ with helpers.vim-plugin;
...
-
-      showCursorLine = mkDefaultOpt {
-        type = types.bool;
-        description = ''
-          Show cursor line
-
-          Default: `true`
-        '';
-      };
+    # TODO introduced 2024-02-19: remove 2024-03-19
```
Comment (traxys) [inline plugins/utils/undotree.nix:16]: 04-19

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/utils/undotree.nix (hunk 2 above)
Comment (GaetanLepage) [reply plugins/utils/undotree.nix]: Done

**Review by traxys — APPROVED** [review]

**Pull request merged**